susemanager-utils/susemanager-sls/src/grains/cpuinfo.py

```diff
@@ -114,3 +116,30 @@ def total_num_cpus():
     sysdev = '/sys/devices/system/cpu/'
     return {'total_num_cpus': len([cpud for cpud in (os.path.exists(sysdev) and os.listdir(sysdev) or list())
                                    if re_cpu.match(cpud)])}
+
+
+def cpu_data():
+    """
+    Returns the cpu model, vendor ID and other data that may not be in the cpuinfo
+    """
+    lscpu = _which_bin(['lscpu'])
+    if lscpu is not None:
+        try:
+            log.debug("Trying lscpu to get CPU data")
+            ret = __salt__['cmd.run_all']('{0} -J'.format(lscpu), output_loglevel='quiet')
+            if ret['retcode'] == 0:
+                data = json.loads(ret["stdout"])
+                name_map = {
+                    "Model name": "cpu_model",
+                    "Vendor ID": "cpu_vendor",
+                    "NUMA node(s)": "cpu_numanodes",
+                    "Stepping": "cpu_stepping",
+                    "Core(s) per socket": "cpu_cores",
+                }
+                values = {name_map[entry["field"][:-1]]: entry["data"] for entry in data.get("lscpu") if entry["field"][:-1] in name_map.keys()}
+                log.debug(values)
+                return values
+            else:
+                log.warn("lscpu does not support -J option")
```

Comment:
Nit: Doe the non-zero return code always mean the `-J` option is not supported? I'd be inclined to say that it can happen in other situations too.

Reply:
I really fail to see what other reason may prevent `lscpu` to work.

Reply:
Do you know `lscpu` that well? Moreover, does it also cover future cases?